doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -49,17 +99,33 @@ By default `runCommand` runs in a stdenv with no compiler environment, whereas [
 ```
 :::
 
-## `runCommandCC` {#trivial-builder-runCommandCC}
+::: {.note}
+`runCommand name derivationArgs buildCommand` is equivalent to
+```nix
+runCommandWith {
+  inherit name derivationArgs;
+  stdenv = stdenvNoCC;
+} buildCommand
+```
+
+Likewise, `runCommandCC name derivationArgs buildCommand` is equivalent to
+```nix
+runCommandWith {
+  inherit name derivationArgs;
+} buildCommand
+```
+
+:::
 
-This works just like `runCommand`. The only difference is that it also provides a C compiler in `buildCommand`'s environment. To minimize your dependencies, you should only use this if you are sure you will need a C compiler as part of running your command.
 
 ## `runCommandLocal` {#trivial-builder-runCommandLocal}
 
-Variant of `runCommand` that forces the derivation to be built locally, it is not substituted. This is intended for very cheap commands (<1s execution time). It saves on the network round-trip and can speed up a build.
+`runCommandLocal :: String -> AttrSet -> String -> Derivation`
```

Comment:
````suggestion
# Type
```
runCommandLocal :: String -> { <type out please> }-> String -> Derivation``
```
````

This is more precise and matches what we have in lib already for all other functions.

The type `AttrSet` doesnt add much value. I'd prefer something like `{ derivationArgs :: { ... }; stdenv :: Derivation; ... }` because that is more concrete

Reply:
When i look at the function definition of `runCommandWith` i see the following:


```nix
  runCommandWith =
    let
      # prevent infinite recursion for the default stdenv value
      defaultStdenv = stdenv;
    in
    {
      # which stdenv to use, defaults to a stdenv with a C compiler, pkgs.stdenv
      stdenv ? defaultStdenv
      # whether to build this derivation locally instead of substituting
    , runLocal ? false
      # extra arguments to pass to stdenv.mkDerivation
    , derivationArgs ? { }
      # name of the resulting derivation
    , name
      # TODO(@Artturin): enable strictDeps always
    }: buildCommand:
```

This would have the following type:

```
runCommandLocal :: String -> {
    name :: name;
    stdenv? :: Derivation;
    runLocal? :: Bool;
    derivationArgs? :: { ... };  <- args forwarded to mkDerivation
  }-> String -> Derivation``
```

If you could use this instead of Attrs i think would be much better.

Reply:
Oh sorry for the confusion. This comment was intended for `runCommandWith`